qiskit_experiments/framework/composite/parallel_experiment.py

```diff
@@ -88,9 +94,35 @@ def circuits(self):
                     sub_circ = sub_circuits[exp_idx][circ_idx]
                     num_clbits = circuit.num_clbits
                     qubits = sub_qubits[exp_idx]
+                    qargs_map = sub_maps[exp_idx]
                     clbits = list(range(num_clbits, num_clbits + sub_circ.num_clbits))
                     circuit.add_register(ClassicalRegister(sub_circ.num_clbits))
-                    circuit.append(sub_circ, qubits, clbits)
+
+                    # Apply transpiled subcircuit
+                    # Note that this assumes the circuit was not expanded to use
+                    # any qubits outside the specified physical qubits
+                    for inst, qargs, cargs in sub_circ.data:
```

Comment:
Can we do this operation on DAG circuit (since operation is written in rust)? Likely the same logic is written here.
https://github.com/Qiskit/qiskit-terra/blob/6d3a4f8f9ebbc55f2f6f6ecf01b49ffed800db53/qiskit/dagcircuit/dagcircuit.py#L717-L737

Reply:
This will be much faster than converting to a DAG and back for any DAG based methods, since that conversion will need to do this iteration anyway.

Reply:
compose for an instruction does additional validation and is much slower than calling `_append` directly when you already have correctly formatted instruction and bit arguments